### PR TITLE
Added tests for RenameVariableToMatchNewTypeRector 

### PR DESCRIPTION
**Pull request by leoloso**
As requested in https://github.com/rectorphp/rector/pull/4879#issuecomment-745132901

However, the tests are not failing in the expected manner:

- Test `with_method_call.php.inc` should not fail, but it does
- Test `with_double_method_call.php.inc` replicates the issue from Rector Rectify. It fails, but it a different way than expected: `$flag` was not renamed as `$bitwiseOr`, and `$magicGet` was left intact

Am I missing something?

### CODE REVIEW

**Comment by leoloso** [comment]
Changes to fix, applied by Rector Rectify, can be seen here: https://github.com/rectorphp/rector/commit/182aef1bbc73966ce3ee74ec951988c52d42fdde

File: rules/naming/tests/Rector/ClassMethod/RenameVariableToMatchNewTypeRector/Fixture/with_double_method_call.php.inc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                $magicGet,
+            ),
+            $magicSet,
+        );
+    }
+
+    protected function createClassConstFetch(string $class, string $constant): ClassConstFetch
+    {
+        return new ClassConstFetch($class, $constant);
+    }
```
Comment (TomasVotruba) [inline rules/naming/tests/Rector/ClassMethod/RenameVariableToMatchNewTypeRector/Fixture/with_double_method_call.php.inc:36]: There is too much going on. Please spilt it to standlone files per case

**Comment by TomasVotruba** [comment]
The test has probably different scope then real file.

Is `RenameVariableToMatchNewTypeRector` the source of origin?

**Comment by leoloso** [comment]
You were right, I got the wrong rule. The actual one is `RenameVariableToMatchMethodCallReturnTypeRectorTest`.

> There is too much going on. Please spilt it to standlone files per case

I had kept the added test so long, to find out what was different. Now that I found out the problem, I made the test just minimal.

**Comment by TomasVotruba** [comment]
@leoloso Thank you :+1:

**Comment by TomasVotruba** [comment]
@samsonasik Could you check this one?

It should be either skipped or better renamed with type suffix, like this:

```diff
-$magicGet = $this->createClassConstFetch('SomeClass', 'MAGIC_GET');
+$magicGetClassConstFetch = $this->createClassConstFetch('SomeClass', 'MAGIC_GET');

-$magicSet = $this->createClassConstFetch('SomeClass', 'MAGIC_SET');
+$magicSetClassConstFetch = $this->createClassConstFetch('SomeClass', 'MAGIC_SET');
```

**Comment by samsonasik** [comment]
I will try.

**Review by simivar — APPROVED** [review]

**Comment by samsonasik** [comment]
fixed.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**